plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/PointColorGradient.tsx

```diff
@@ -1,13 +1,163 @@
+import {
+  TextField,
+  ColorField,
+} from "@web/extensions/sidebar/core/components/content/common/DatasetCard/Field/common";
+import { Wrapper } from "@web/extensions/sidebar/core/components/content/common/DatasetCard/Field/commonComponents";
+import { generateColorGradient } from "@web/extensions/sidebar/utils/color";
+import { isEqual, omit } from "lodash";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
 import { BaseFieldProps } from "../types";
 
+const generateConditions = (
+  field?: string,
+  min?: number,
+  max?: number,
+  step?: number,
+  startColor?: string,
+  endColor?: string,
+) => {
+  if (!field || !min || !max || !step || !startColor || !endColor || min >= max || step >= max) {
+    return [];
+  }
+
+  const values = [];
+  for (let i = min; i <= max; i += step) values.push(i);
+  const colors = generateColorGradient(startColor, endColor, values.length);
+  const fieldName = "${" + String(field) + "}";
+  return values
+    .map((value, index) => [`${fieldName} >= ${String(value)}`, `'${String(colors[index])}'`])
+    .reverse();
+};
+
 const PointColorGradient: React.FC<BaseFieldProps<"pointColorGradient">> = ({
   value,
   editMode,
+  isActive,
   onUpdate,
 }) => {
-  // remember to update the BaseFieldProps type!
-  console.log(value, editMode, onUpdate);
-  return <div>PointColor</div>;
+  const [colorGradient, setColorGradient] = useState(value);
+
+  const handleFieldChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setColorGradient({ ...colorGradient, field: e.target.value });
+    },
+    [colorGradient],
+  );
+
+  const handleMinValueUpdate = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const min = !isNaN(parseFloat(e.currentTarget.value)) ? parseFloat(e.currentTarget.value) : 1;
+      setColorGradient({ ...colorGradient, min });
+    },
+    [colorGradient],
+  );
+
+  const handleMaxValueUpdate = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const max = !isNaN(parseFloat(e.currentTarget.value)) ? parseFloat(e.currentTarget.value) : 1;
+      setColorGradient({ ...colorGradient, max });
+    },
+    [colorGradient],
+  );
+
+  const handleStartColorUpdate = useCallback(
+    (color: string) => {
+      // colors should be hex for now
```

Comment:
Can you look in to [this page](https://www.w3schools.com/jsref/jsref_getcomputedstyle.asp) for perhaps supporting red, green, pink, etc?

Reply:
I saw this yesterday but I thought that we can not use window object.
```ts
window.getComputedStyle(element, null);
```
I will check again.